crates/solana-gateway-task-processor/src/component.rs

```diff
@@ -140,26 +153,44 @@ struct ConfigMetadata {
 async fn process_task(
     keypair: &Keypair,
     solana_rpc_client: &RpcClient,
-    task: TaskItem,
+    mut amplifier_client: relayer_amplifier_api_integration::AmplifierCommandClient,
+    task_item: TaskItem,
     metadata: &ConfigMetadata,
 ) -> eyre::Result<()> {
     use amplifier_api::types::Task::{Execute, GatewayTx, Refund, Verify};
     let signer = keypair.pubkey();
     let gateway_root_pda = metadata.gateway_root_pda;
 
-    match task.task {
+    match task_item.task {
         Verify(_verify_task) => {
             tracing::warn!("solana blockchain is not supposed to receive the `verify_task`");
         }
         GatewayTx(task) => {
             gateway_tx_task(task, gateway_root_pda, signer, solana_rpc_client, keypair).await?;
         }
         Execute(task) => {
+            let source_chain = task.message.source_chain.clone();
+            let message_id = task.message.message_id.clone();
+
             // communicate with the destination program
-            execute_task(task, metadata, signer, solana_rpc_client, keypair)
+            if let Err(error) = execute_task(task, metadata, signer, solana_rpc_client, keypair)
```

Comment:
We need to update `send_tx` to return the error even when not irrecoverable and parse the errors here to return the appropriate event.